srcalc/src/test/java/gov/va/med/srcalc/vista/RpcVistaPatientDaoTest.java

```diff
@@ -202,4 +205,29 @@ public final void testNoHealthFactors()
         final Patient patient = dao.getPatient(PATIENT_DFN);
         assertEquals(Collections.<HealthFactor>emptyList(), patient.getHealthFactors());
     }
+    
+    @Test
+    public final void testMedicationsValid()
+    {
+        final VistaProcedureCaller caller = mockVistaProcedureCaller();
+        when(caller.doRpc(
+                RADIOLOGIST_DUZ,
+                RemoteProcedure.GET_ACTIVE_MEDICATIONS,
+                String.valueOf(PATIENT_DFN), "", ""))
+                .thenReturn(VALID_ACTIVE_MEDICATIONS);
+        final RpcVistaPatientDao dao = new RpcVistaPatientDao(caller, RADIOLOGIST_DUZ);
+        final Patient patient = dao.getPatient(PATIENT_DFN);
+        final List<String> expectedMedications = ImmutableList.of(
+                "METOPROLOL TARTRATE 50MG TAB", "SIMVASTATIN 40MG TAB");
+        assertEquals(expectedMedications, patient.getActiveMedications());
+    }
+    
+    @Test
+    public final void testNoMedications()
+    {
+        final VistaProcedureCaller caller = mockVistaProcedureCaller();
+        final RpcVistaPatientDao dao = new RpcVistaPatientDao(caller, RADIOLOGIST_DUZ);
+        final Patient patient = dao.getPatient(PATIENT_DFN);
+        assertEquals(Collections.<String>emptyList(), patient.getActiveMedications());
+    }
```

Comment:
These are good tests, but I suggest one more (also for health factors): test our exception handling by having VistA return something invalid.